ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OAlgoCommonParams.scala

```diff
@@ -20,70 +20,54 @@ import ai.h2o.sparkling.ml.params.H2OAlgoParamsHelper.getValidatedEnumValue
+  protected def paramTag: ClassTag[P]
+
+  protected var parameters: P = paramTag.runtimeClass.newInstance().asInstanceOf[P]
 
   //
   // Param definitions
   //
-  protected final val modelId = new NullableStringParam(
-    this,
+  protected final val modelId = nullableStringParam(
     "modelId",
     "An unique identifier of a trained model. If the id already exists, a number will be appended to ensure uniqueness.")
-  private val keepCrossValidationPredictions = booleanParam("keepCrossValidationPredictions")
-  private val keepCrossValidationFoldAssignment = booleanParam("keepCrossValidationFoldAssignment")
-  private val parallelizeCrossValidation = booleanParam("parallelizeCrossValidation")
-  private val distribution = stringParam("distribution")
+  private val distribution = stringParam("distribution", "Distribution function")
 
   //
   // Default values
   //
-  setDefault(
-    modelId -> null,
-    keepCrossValidationPredictions -> parameters._keep_cross_validation_predictions,
-    keepCrossValidationFoldAssignment -> parameters._keep_cross_validation_fold_assignment,
-    parallelizeCrossValidation -> parameters._parallelize_cross_validation,
-    distribution -> parameters._distribution.name())
+  setDefault(modelId -> null, distribution -> parameters._distribution.name())
 
   //
   // Getters
   //
   def getModelId(): String = $(modelId)
 
-  def getKeepCrossValidationPredictions(): Boolean = $(keepCrossValidationPredictions)
-
-  def getKeepCrossValidationFoldAssignment(): Boolean = $(keepCrossValidationFoldAssignment)
-
-  def getParallelizeCrossValidation(): Boolean = $(parallelizeCrossValidation)
-
   def getDistribution(): String = $(distribution)
 
   //
   // Setters
   //
   def setModelId(id: String): this.type = set(modelId, id)
 
-  def setKeepCrossValidationPredictions(value: Boolean): this.type = set(keepCrossValidationPredictions, value)
-
-  def setKeepCrossValidationFoldAssignment(value: Boolean): this.type = set(keepCrossValidationFoldAssignment, value)
-
-  def setParallelizeCrossValidation(value: Boolean): this.type = set(parallelizeCrossValidation, value)
-
   def setDistribution(value: String): this.type = {
     set(distribution, getValidatedEnumValue[DistributionFamily](value))
   }
 
-  private[sparkling] def getH2OAlgorithmParams(): Map[String, Any] = {
-    Map(
-      "weights_column" -> getWeightCol(),
-      "nfolds" -> getNfolds(),
-      "fold_column" -> getFoldCol(),
-      "keep_cross_validation_predictions" -> getKeepCrossValidationPredictions(),
-      "keep_cross_validation_fold_assignment" -> getKeepCrossValidationFoldAssignment(),
-      "parallelize_cross_validation" -> getParallelizeCrossValidation(),
-      "seed" -> getSeed(),
-      "distribution" -> getDistribution())
+  private[sparkling] override def getH2OAlgorithmParams(): Map[String, Any] = {
+    super.getH2OAlgorithmParams() ++
+      Map(
+        "weights_column" -> getWeightCol(),
```

Comment:
Maybe could we add comment why these can't be auto-generated? EDIT: I see they are mentioned in `IgnoredParams` class, that is fine

Also I see that `weights_column` & `fold_column` have different naming, but couldn't we also auto-generate `seed`, `distribution` & `nfolds` ?

Reply:
These parameters are defined in `H2OCommonParams` and utilized also by `GridSearch` and `AutoML`. Once parameters for `GridSearch` and `AutoML` are generated as well, i think we could remove `H2OCommonParams` class and change the hierarchy of traits a bit.